formats/polus-tabular-to-feather-plugin/src/main.py

```diff
@@ -0,0 +1,128 @@
+        logger.info('Saving csv file ' + file_name)
+        #Export the file as csv
+        os.chdir(outDir)
+        export_csv = data.to_csv (r'%s.csv'%file_name, index = None, header=True,encoding='utf-8-sig')
+        return export_csv
+    
+def df_toFeather(file,outDir):
+        """Convert a csv or fcs file to Arrow feather file. Includes code from polus-fcs-to-csv-converter plugin
+                
+            Args:
+                file (str): Path to the directory containing the csv file.
+                outDir (str): Path to the directory to save feather file.
+                    
+            Returns:
+                Feather format file.
+                    
+            """
+        file_name = Path(file).stem
+        feather_filename = file_name + '.feather'
+        
+        logger.info('Started converting the csv file ' + file_name)
+        df = vaex.from_csv(file)
+        logger.info('writing file...')
+        os.chdir(outDir)
+        df.export_feather(feather_filename,outDir)
+
+def main(inpDir: Path,
+            outDir: Path,
+            ) -> None:
+        """ Main execution function
+        
+        All functions in your code must have docstrings attached to them, and the
+        docstrings must follow the Google Python Style:
+        https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html
+        """
+        
+        # pattern = filePattern if filePattern is not None else '.*'
+        # fp = filepattern.FilePattern(inpDir,pattern)
+        
+        
+        logger.info('outDir = {}'.format(outDir))
+        logger.info('inpDir = {}'.format(inpDir))
+        
+        # Surround with try/finally for proper error catching
+        try:
+            
+            #List the files in the directory
+            logger.info('Checking for .csv or .fcs files in the directory ')
+            fcs_filelist = list(Path(inpDir).glob('*.fcs'))
+            if not fcs_filelist:
+                raise FileNotFoundError('No .fcs files were found in the directory. Please check file directory.' )
+        
+            for each_file in fcs_filelist:
+                #Convert fcs to csv
+                logger.info("Checking for fcs files...")
+                file_ext = Path(each_file).suffix
+                if file_ext == '.fcs':
+                    logger.info('Converting fcs file to csv ')
+                    each_file = fcs_csv(each_file, inpDir)
+        finally:    
+            logger.info('Checking for .csv files in the directory ')
+            filelist = list(Path(inpDir).glob('*.csv'))
+            if not filelist:
+                raise FileNotFoundError('No .csv files were found in the directory. Please check file directory.' )
+                
+            for each_file in filelist:
+                #Read the csv file into feather
+                df_toFeather(each_file,outDir)
+                
+            logger.info("Finished all processes!")
```

Comment:
I don't see the value in using `try/catch` here. Usually this is to catch known and well defined errors, but it looks like it's being used as a switch to convert non-fcs files.

Also, we should try to convert everything that vaex supports (csv, parquet, hdf5, etc) and fcs. It looks like the current code only supports fcs and csv.